src/Features/Core/Portable/CodeFixes/Suppression/ConfigureSeverityLevelCodeAction.cs

```diff
@@ -0,0 +1,284 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Suppression
+{
+    internal sealed class ConfigureSeverityLevelCodeAction : CodeAction.CodeActionWithNestedActions
+    {
+        public ConfigureSeverityLevelCodeAction(
+            Diagnostic diagnostic,
+            ImmutableArray<CodeAction> nestedActions)
+            : base(FeaturesResources.Configure_severity_level_via_editorconfig_file,
+            ImmutableArray<CodeAction>.CastUp(nestedActions), isInlinable: false)
+        {
+        }
+
+        internal static async Task<Solution> ConfigureEditorConfig(
+            string severity,
+            Diagnostic diagnostic,
+            Project project,
+            Dictionary<string, Option<CodeStyleOption<bool>>> languageSpecificOptions,
+            Dictionary<string, Option<CodeStyleOption<ExpressionBodyPreference>>> expressionOptions,
+            string language,
+            CancellationToken cancellationToken)
+        {
+            var solution = project.Solution;
+            var editorconfig = FindOrGenerateEditorconfig();
+            var result = await editorconfig.GetTextAsync().ConfigureAwait(false);
+
+            var headers = new Dictionary<string, TextLine>();
+            string mostRecentHeader = null;
+            var lines = result.Lines;
+
+            var name = FindRuleName();
+            if (name != null)
+            {
+                // First we check if the rule already exists in the file; if it does, we replace it
+                if (CheckIfRuleExistsAndReplaceInFile())
+                {
+                    return solution;
```

Comment:
if hte rule exists... why would we offer the feature in the first place?   That concerns be.  That means that even if the user has a rule for this feature, we'll be bloating the lightbulb menu for all future invocations?  I don't think that's an ok user experience.

Reply:
I don't understand this pattern.  All codepaths end up returning 'solution'.  so what actually causes anything to change?

Reply:
1. Based on customer interviews Kasey and I conducted, some customers indicated that they wanted to change the severity level of rules depending on what stage of the code development process they were in. So for example, in order for people to pay greater attention to fixing a diagnostic, they may now want a rule that was previously configured as a warning to now be configured as an error.

2. The change is caused by calling solution.WithAdditionalDocument(…) and ultimately returning the solution returned from the method.

Reply:
sorry, my point wasn't clear.  What i was trying to say was: this makes it look as if you're returning the 'solution' *untouched*.  That's because you actually mutate it in non-local ways because of your use of local functions.  That makes the code actually look like it's doing something else.

Reply:
See: https://github.com/dotnet/roslyn/pull/29332#discussion_r212139780